File: Excel/634 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/634 Array Equality.xlsx"
input = read_excel(path, range = "A2:B10")
test  = read_excel(path, range = "D2:E7")

result = input %>%
  mutate(set_array1 = map(Array1, ~ sort(unique(strsplit(.x, ",")[[1]]))),
         set_array2 = map(Array2, ~ sort(unique(strsplit(.x, ",")[[1]])))) %>%
  mutate(result = map2(set_array1, set_array2, ~ .x %>% setequal(.y))) %>%
  filter(result == TRUE) %>%
  select(Array1, Array2)

all.equal(result, test, check.attributes = FALSE)
#> [1] TRUE

File: Excel/634 Challenge.py
import pandas as pd

path = "634 Array Equality.xlsx"
input = pd.read_excel(path,  usecols="A:B", skiprows=1, nrows=9)
test = pd.read_excel(path,  usecols="D:E", skiprows=1, nrows=5).rename(columns=lambda x: x.split('.')[0])

def split_sort_unique(s):
    return sorted(set(s.split(',')))

input['set_array1'] = input['Array1'].apply(split_sort_unique)
input['set_array2'] = input['Array2'].apply(split_sort_unique)
input['result'] = input.apply(lambda row: row['set_array1'] == row['set_array2'], axis=1)
result = input[input['result'] == True][['Array1', 'Array2']].reset_index(drop=True)

print(result.equals(test)) #Free